### DISCUSSION TITLE
"Foreign" films.

### DISCUSSION THREAD

**Discussion by subhappy** [ptr727/PlexCleaner#134] [Q&A] [answered]
Hi there! Being a cinephile, I have a bunch of "foreign" films, where the original soundtrack could be in German, French, Indonesian, Mandarin, Cantonese, etc.
Do I need to define all of these so Plexcleaner doesn't remove them, or what would happen if I run Plexcleaner with default settings?

Is there a way to do a dry-run without any destructive changes?

**Comment by ptr727** [comment, marked as answer]
You need to specify all languages you want to keep, or you can set `RemoveUnwantedLanguageTracks: false` to not remove any.

Use `--testnomodify` as [documented](https://github.com/ptr727/PlexCleaner#process-media-files) an option for the `process` command.  

I would suggest you test using a copy of your files until you are satisfied with the results before you do bulk processing.  
Or if you use e.g. ZFS take a snapshot of your media volume that can easily be reverted.

**Reply by subhappy** [reply]
Thanks for the prompt reply, I see that I overlooked the testnomodify option.

But is there any way to just keep the "original" audio, the default track, and dump the rest? I.e. for a German-language film (Das Boot) I would like to keep it, but I can live without Die Hard being dubbed to German.

**Reply by ptr727** [reply]
There is no option to keep just the "default".  
To my knowledge there is also no metadata to describe "original", maybe if the video track language was set, and then assume the video track language is "original", but it is very rare to see video tracks with a language set.  
Also keep in mind that I have seen many instances where there is no "default" track marker, and players would then match by user language or first track.

**Reply by subhappy** [reply]
Ah, then I guess the first track would often be the correct one. Thanks for the info.

**Reply by ptr727** [reply]
I noticed there is a newish `FlagOriginal` tag.
I created an [enhancement](https://github.com/ptr727/PlexCleaner/issues/147) to track possibly adding logic to always keep the original tracks.
I don't think I've seen example media with this tag yet, if you have, do let me know please.

**Reply by ptr727** [reply]
You can try out v3 beta (develop branch), I added an option to keep the original language if it was tagged in metadata.

**Comment by sajanvox30** [comment]
By default, Plexcleaner does not remove foreign language audio tracks unless they are flagged as extras or commentary tracks. So, you should not have to define all of the languages in your media library.

However, if you have made changes to the default settings, such as checking the "Remove audio tracks in languages not listed in 'Audio Languages'" option, then you may need to ensure that all of the foreign language tracks are listed in the "Audio Languages" section of your Plex library.

To perform a dry-run without making any destructive changes, you can use the "--dry-run" option when running Plexcleaner. This will allow you to see what changes would be made without actually deleting any files. For example, to perform a dry-run on a library called "Movies", you can use the following command:

**Comment by subhappy** [comment]
Great, I'll try. It uses the flagOriginal tag?

Lars Augensen
40224293
